ui/pages/confirmations/components/confirm/utils.ts

```diff
@@ -17,3 +17,14 @@ export const getConfirmationSender = (
 
   return { from };
 };
+
+export const formatNumber = (value: number, decimals: number) => {
+  if (value === undefined) {
+    return value;
+  }
+  const formatter = new Intl.NumberFormat('en-US', {
```

Comment:
I've since looked into the code and I'd like to surface 2 other `formatNumber` methods I've found used in our repo. Each have their own nuances. We utilize Intl.NumberFormat for each of the methods, but differently. 

`app/scripts/controllers/push-platform-notifications/utils/get-notification-data.ts`:
- uses abbreviation / notation (K, M, B, T)
- uses ellipsis
- does not support locales; Uses en-US

`ui/pages/confirmations/components/simulation-details/formatAmount.ts`:
- neither abbreviated nor uses ellipsis
- displays all left numbers if value is >= 0 
- rounds decimals based on desired precision
- supports locales which if I'm not mistaken only utilizes the benefit of commas vs decimals. Symbols are handled outside of the method.

since the related [issue](https://github.com/MetaMask/MetaMask-planning/issues/2649) ticket mentions:
> This is already done for simulations UI that exists on transactions confirmations in case there's logic we can re-use from there.

I'd suggest we extract the simulation details `formatAmount` method to be reusable across the repo. We could update the method to specify the precision number / significant decimal places. This suggestion would apply more cohesion across our app, reduce `formatNumber` variants, and support i18n locale decimals/commas.

Reply:
regarding the last suggestion to remove formatNumber in favor of reusing the simulation's formatAmount method, I see you are doing in in a subsequent PR here https://github.com/MetaMask/metamask-extension/pull/25438/files#diff-27e9fac2551026bbb4a89ea07db94a27210316d346fed5e3234e696601e425b5L21. 👍🏼  

other comment to support locale still applies